lib/fswatcher/fswatcher.go

```diff
@@ -61,16 +66,14 @@ func NewFsWatcher(folderPath string, folderID string, ignores *ignore.Matcher) *
 	}
 }
 
-func (watcher *FsWatcher) StartWatchingFilesystem() (<-chan FsEventsBatch, error) {
+func (watcher *FsWatcher) StartWatchingFilesystem() (chan FsEventsBatch, error) {
 	fsEventChan, err := watcher.setupNotifications()
 	if err == nil {
 		watcher.WatchingFs = true
 		watcher.fsEventChan = fsEventChan
 		go watcher.watchFilesystem()
 	}
-	notifyModelChan := make(chan FsEventsBatch)
-	watcher.notifyModelChan = notifyModelChan
```

Comment:
Couldn't the two lines above just be moved to the beginning of this function to both eliminate the race and keep the unidirectional channels?